Bottom: 8d2384342bc59266a2e892c90b4eb160fd8a2f99
Top:    03c812b7086977876a0ae6ac103406e3cabb91b8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 02:55:59 +0100

Refresh of draw-c-use-drawppv-where-we-ca

---

diff --git a/src/draw.c b/src/draw.c
index 5a9bb1b..b5be3dc 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -376,25 +376,10 @@ DrawEverything (BoxTypePtr drawn_area)
       int group = drawn_groups[i];
 
       if (gui->set_layer (0, group, 0))
-	{
-	  if (DrawLayerGroup (group, drawn_area) && !gui->gui)
-	    {
-	      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_callback, NULL);
-	      r_search (PCB->Data->via_tree, drawn_area, NULL, via_callback, NULL);
-	      /* draw element pads */
-	      if (group == component || group == solder)
-		{
-                  side = (group == solder) ? SOLDER_LAYER : COMPONENT_LAYER;
-		  r_search (PCB->Data->pad_tree, drawn_area, NULL, pad_callback, &side);
-		}
-
-	      /* draw holes */
-	      plated = -1;
-	      r_search (PCB->Data->pin_tree, drawn_area, NULL, hole_callback, &plated);
-	      r_search (PCB->Data->via_tree, drawn_area, NULL, hole_callback, &plated);
-	    }
-	}
+        if (DrawLayerGroup (group, drawn_area) && !gui->gui)
+          DrawPPV (group, drawn_area);
     }
+
   if (TEST_FLAG (CHECKPLANESFLAG, PCB) && gui->gui)
     return;
